cli/emit.rs

```diff
@@ -378,57 +298,50 @@ pub struct CheckOptions {
   pub type_check_mode: TypeCheckMode,
   /// Set the debug flag on the TypeScript type checker.
   pub debug: bool,
-  /// If true, any files emitted will be cached, even if there are diagnostics
-  /// produced. If false, if there are diagnostics, caching emitted files will
-  /// be skipped.
-  pub emit_with_diagnostics: bool,
   /// The module specifier to the configuration file, passed to tsc so that
   /// configuration related diagnostics are properly formed.
   pub maybe_config_specifier: Option<ModuleSpecifier>,
   /// The derived tsconfig that should be used when checking.
   pub ts_config: TsConfig,
   /// If true, `Check <specifier>` will be written to stdout for each root.
   pub log_checks: bool,
-  /// If true, valid existing emits and `.tsbuildinfo` files will be ignored.
+  /// If true, valid `.tsbuildinfo` files will be ignored and type checking
+  /// will always occur.
   pub reload: bool,
-  pub reload_exclusions: HashSet<ModuleSpecifier>,
```

Comment:
Need to bring this back as well

Reply:
It seems to add too much complexity. I think we should just always type check if someone provides `--reload`.

Reply:
(sorry I hadn't seen https://github.com/denoland/deno/pull/15118#discussion_r918421687)

It makes it so the `--reload` flag is only applied once-per-module-per-process. We previously had cases modules would be redownloaded for every dynamic import and worker if `--reload` was passed. Omitting it here would only do that for type checking which would have significantly less impact, but it's probably sound to have the once-per-module-per-process guarantee anywhere `--reload` is processed.

Reply:
Ok, it looks like we can push this up to proc_state. I'll do that now.